src/syncData.ts

```diff
@@ -54,7 +52,15 @@ export const initSync = (
   const tableName = options.tableName || collection.collectionName.toLowerCase()
   const qualifiedName = `"${schemaName}"."${tableName}"`
   // Initialize sync
-  const sync = mongoChangeStream.initSync<Events>(redis, collection, options)
+  const sync = mongoChangeStream.initSync<Events>(redis, collection, {
+    ...options,
+    retry: {
+      shouldRetry(error) {
+        // Don't retry if we get this exception
+        return !error.message.includes('DuplicateKeyException')
+      },
+    },
```

Comment:
Would it make sense to flip these, so that the caller can override the `retry` property if they want?

Reply:
Not sure what you mean. Can you elaborate more? For context, all options under the `retry` field, including `shouldRetry`, are directly passed through to `p-retry` without modification.

Reply:
Currently the order is:

```typescript
{
  ...options,
  retry: {
    shouldRetry() { /* ... */ }
  }
}
```

Which means if the consumer of the mongo2crate library decides to pass in their own `options.retry`, it will be ignored and replaced with the retry options above.

If we instead flip it so that it looks like this:

```typescript
{
  retry: {
    shouldRetry() { /* ... */ }
  },
  ...options
}
```

This would provide the retry options above by default, but if desired, the caller could include `options.retry` to override them, e.g. if they have specific logic about when to retry.